src/Knp/DictionaryBundle/Dictionary/CallableDictionary.php

```diff
@@ -131,16 +133,17 @@ public function unserialize($serialized)
     {
         $data = unserialize($serialized);
 
-        $this->name   = $data['name'];
-        $this->values = $data['values'];
+        $this->name        = $data['name'];
+        $this->values      = $data['values'];
+        $this->initialized = true;
```

Comment:
What about `$this->initialized = empty($data['values']);`
We don't know if the dictionary has been initialized before its serialization.